core/src/main/scala/org/apache/spark/deploy/worker/DriverRunner.scala

```diff
@@ -171,16 +174,20 @@ private[deploy] class DriverRunner(
   private[worker] def prepareAndRunDriver(): Int = {
     val driverDir = createWorkingDirectory()
     val localJarFilename = downloadUserJar(driverDir)
+    val resourceFile = ResourceUtils.prepareResourceFile(SPARK_DRIVER_PREFIX, resources, driverDir)
 
     def substituteVariables(argument: String): String = argument match {
       case "{{WORKER_URL}}" => workerUrl
       case "{{USER_JAR}}" => localJarFilename
       case other => other
     }
 
+    // config resource file for driver, which would be used to load resources when driver starts up
+    val javaOpts = driverDesc.command.javaOpts ++
+      Seq(s"-D${DRIVER_RESOURCES_FILE.key}=${resourceFile.getAbsolutePath}")
```

Comment:
always sending this, vs only when actually needs resources?  I assume the file is empty if it doesn't need any?

Reply:
Good idea.